bootstrap.sh

```diff
@@ -27,71 +45,124 @@ PermitRootLogin no
 EOF
-
-cat > ~/.gemrc <<EOF
+  
+  echo "--- add 'gem: --no-document' so gem installs don't include documentation"
+  cat > ~/.gemrc <<EOF
 gem: --no-document
 EOF
-
-sudo softwareupdate --schedule off
-
-ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
-
-brew update
-
-brew install git wget mercurial xctool node coreutils postgresql postgis sqlite go gpg carthage md5deep
-
-echo "source /usr/local/opt/nvm/nvm.sh" >> ~/.bashrc
-
-brew install caskroom/cask/brew-cask
-
-brew cask install xquartz
-brew cask install java
-brew cask install oclint
-brew cask install rubymotion
-
-brew install maven
-
-gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
-\curl -sSL https://get.rvm.io | bash -s stable
-source "$HOME/.rvm/scripts/rvm"
-rvm install jruby-1.7.19
-rvm install 1.9.3-p551
-rvm install 2.1.5
-rvm install 2.2.1
-rvm install 2.0.0-p643
-
-rvm alias create default 2.0.0-p643
-
-rvm all do gem install nomad-cli cocoapods bundler rake xcpretty fastlane
-
-pod setup
-
-sudo tee /etc/launchd.conf <<EOF
+  
+  echo "--- Turn off automatic software updating"
+  sudo softwareupdate --schedule off
+  
+  echo "--- Install/upgrade brew."
+  brew upgrade || ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+  
+  echo "--- Install tools with brew"
+  ## homebrew fun
+  brew update
+  
+  for PKG in "${BREW_PKGS[@]}"; do
+    if [[ ! $(brew list | grep $PKG) ]]; then
+      brew install $PKG
+    else 
+     echo "$PKG present"
+    fi
+  done
+
+  echo "--- Install brew-cask"
+  brew install caskroom/cask/brew-cask
+  
+  echo "--- Install tools with brew-cask"
+  brew cask install xquartz
+  brew cask install java
+  brew cask install oclint
+  brew cask install rubymotion
+  # To prevent RubyMotion permission errors because `sudo motion update` created
+  # this
+  mkdir -p ~/Library/RubyMotion
+
+  
+  echo "--- Install Maven now that we have Java."
+  brew install maven
+  ## brew fun end
+  
+  echo "--- Update RubyMotion"
+  sudo motion update
+
+  # nvm
+  echo "--- Install nvm"
+  if [[ ! -d $HOME/.nvm ]]; then
+    wget -qO- "https://raw.githubusercontent.com/creationix/nvm/$NVM_VERSION/install.sh" | bash
+  fi
+  
+  source $HOME/.nvm/nvm.sh
+  
+  for VER in "${NODE_VERSIONS[@]}"; do
+    if [[ ! $(nvm list | grep $VER) ]]; then
+      nvm install $VER
+    else
+      echo "$VER present"
+    fi
+  done
+  
+  nvm list
+  # end nvm
+
+  echo "--- Install rvm"
+  if [[ ! -d $HOME/.rvm ]]; then
+    gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
+    curl -sSL https://get.rvm.io | bash -s stable
+  fi
+  source "$HOME/.rvm/scripts/rvm"
+
+  echo "--- Install rubies with rvm"
+  for RUBY in "${RUBIES[@]}"; do
+    if [[ ! $(rvm list | grep $RUBY) ]]; then
+      rvm install $RUBY
+    else
+      echo "$RUBY fine"
+    fi
+  done
+
+  rvm alias create default 2.0.0-p643
```

Comment:
Should we extract this to a variable as well and put it with the `RUBIES` var near the top?